backend/src/app/resolvers/site/site.resolver.ts

```diff
@@ -62,6 +63,78 @@ export class SiteResolver {
     return this.siteService.findAll();
   }
 
+  /**
+   * Find sites where search parameter matches a site id or address
+   * @param searchParam search parameter
+   * @param page page number
+   * @param pageSize size of the page
+   * @returns sites where id or address matches the search param along with pagination params
+   */
+  @Query(() => SearchSiteResponse, { name: 'searchSitesForAuthenticatedUsers' })
+  async searchSitesForAuthenticatedUsers(
+    @AuthenticatedUser() userInfo,
```

Comment:
Could you explain why we're creating a separate query that's almost completely identical to `searchSites`? I don't see a reason we can't add  @AuthenticatedUser() to the original query

Reply:
searchSitesForAuthenticatedUsers is only for authenticated users  , because when you use Unprotected decorator for a controller, @AuthenticatedUser()  returns null basically it skips the pipeline. So basically to handle searches for public / logged in users we need two different methods

Reply:
I see what you mean, but creating two separate queries (and considering how massive they are) is not the answer here. I can look into extending the auth guards we have set up to have optional auth instead of outright skipping it later today.

Reply:
This just a graphql endpoint service layer remains the same.